packages/core-entities/src/models/views/YieldViews.ts

```diff
@@ -430,6 +444,188 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
     }
   };
 
+  const getUniqueUnderlyingSymbols = (productGroupData: any[][]) => {
+    const uniqueUnderlyingSymbols = productGroupData
+      .flat()
+      .map((item) => item.underlying?.symbol)
+      .filter((symbol): symbol is string => symbol !== undefined)
+      .reduce<string[]>((acc, symbol) => {
+        if (!acc.includes(symbol)) {
+          acc.push(symbol);
+        }
+        return acc;
+      }, []);
+
+    return uniqueUnderlyingSymbols;
+  };
+
+  const getDefaultHighestAPYSymbol = (productGroupData: any[][]) => {
+    const highestTotalAPYBySymbol: Record<string, number> = {};
+    productGroupData.flat().forEach((item) => {
+      const symbol = item?.underlying?.symbol;
+      if (symbol) {
+        const totalAPY = item.apy?.totalAPY || 0;
+
+        if (!highestTotalAPYBySymbol[symbol]) {
+          highestTotalAPYBySymbol[symbol] = totalAPY;
+        } else {
+          highestTotalAPYBySymbol[symbol] += totalAPY;
+        }
+      }
+    });
+
+      let highestAPYSymbol = '';
+      let highestAPYValue = 0;
+      for (const data in highestTotalAPYBySymbol) {
+        if (highestTotalAPYBySymbol[data] > highestAPYValue) {
+          highestAPYValue = highestTotalAPYBySymbol[data];
+          highestAPYSymbol = data;
+        }
+      }
+
+      return highestAPYSymbol;
+  };
+
+  const getPortfolioStateZeroBorrowData = () => {
+    const group = ['fCash', 'PrimeDebt'];
+    const productGroupData = group.map((group) => {
+      return getTokensByType(group)
+        .filter((data) => data?.isFCashDebt || data?.tokenType === 'PrimeDebt')
```

Comment:
Unclear why this filter is necessary? Shouldn't this already be done in getTokensByType